### ISSUE TITLE
Ability to invalidate buffer contents on release

### ISSUE THREAD

**Issue by DaveCTurner** [netty/netty#14459]
A feature request to complement `io.netty.util.ResourceLeakDetector` and friends: could we have an `io.netty.buffer.ByteBufAllocator` implementation which creates buffers who invalidate their contents on release?

We (Elasticsearch) run our tests with `-Dio.netty.noResourceLeakDetection=paranoid` to get confidence that we have no buffer leaks, but we also worry about the converse problem of a premature `release()` while a buffer is still in use. This kind of bug gets detected eventually but experience shows it doesn't come up in tests so reliably. We're pretty sure that if we filled buffers with junk before deallocating them then we'd notice any such bugs much more quickly.

I'm imagining it'd wrap another `ByteBufAllocator` with something that delegates to the inner allocator to do the actual allocation, but then adds invalidate-on-deallocate wrappers to all the buffers it produces. Filling with zeroes would do for us. Note that this is only something we'd use in tests, so we are not particularly concerned about the performance impact it might have.

Relates https://github.com/elastic/elasticsearch/pull/116115

**Comment by chrisvest** [comment]
Our bounds checks involve a call to `ByteBuf.ensureAccessible()`. Are you find a way past that check?

**Comment by DaveCTurner** [comment]
Yes, e.g [here](https://github.com/elastic/elasticsearch/blob/91474289122fc610b457b2836f6daafb60e44bd7/modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Utils.java#L126-L127) we use `io.netty.buffer.ByteBuf#array` to re-wrap the underlying `byte[]` in a different abstraction over the same bytes.
